Unpinned importlib-metadata constraint

UsamaSadiq opened the pull request:
- `importlib-metadata<2.0.0` was causing conflicts in running `make upgrade` with `python3.5`.
- Unpinned `importlib-metadata` constraint after fixing the issue in https://github.com/pypa/virtualenv/pull/1953 and https://github.com/tox-dev/tox/pull/1682.

regisb commented:
Hi @UsamaSadiq! I checked your other PRs but I don't understand what problem this is fixing. You mention [here](https://github.com/pypa/virtualenv/pull/1953) that "On Python<3.8, importlib-metadata==2.0.0 causes dependency conflicts because of the pinned constraint python<3.8". Did you mean instead: "... because of the pinned constraint python>=3.8"?

Maybe if you could paste here the error stacktrace that would help clarify the issue?

UsamaSadiq commented:
> Hi @UsamaSadiq! I checked your other PRs but I don't understand what problem this is fixing. You mention [here](https://github.com/pypa/virtualenv/pull/1953) that "On Python<3.8, importlib-metadata==2.0.0 causes dependency conflicts because of the pinned constraint python<3.8". Did you mean instead: "... because of the pinned constraint python>=3.8"?
> 
> Maybe if you could paste here the error stacktrace that would help clarify the issue?

Hi @regisb, `tox` and `virtualenv` both had the constraint `importlib-metadata>=0.12,<2; python<3.8` which was causing conflicts in the builds when ran with `python3.5` or `python3.6`. We had to wait for the fix in both packages so we added the constraint in the meanwhile. Now that constraint has been updated in both `tox` and `virtualenv` so I've removed the pin from the repo too. I hope this clarifies the situation.

regisb commented:
Ok thanks!
